src/pages/TokenDetails/index.tsx

```diff
@@ -123,7 +145,10 @@ export default function TokenDetails() {
         <>
           <TokenDetail address={tokenAddress} />
           <RightPanel>
-            <Widget defaultToken={token ?? undefined} />
+            <Widget
+              defaultToken={token ?? undefined}
+              onReviewSwapClick={shouldShowSpeedbump ? onSwapReview : undefined}
```

Comment:
nit: I think it makes more sense to put this logic in `onSwapReview`, so that the callback immediately returns `true` (or `undefined`) if `!shouldShowSpeedbump`.